File: src/main/java/com/undsf/mcmodmgr/curseforge/models/Pagination.java
package com.undsf.mcmodmgr.curseforge.models;

import com.fasterxml.jackson.annotation.JsonProperty;

public class Pagination {
    @JsonProperty
    public int index;

    @JsonProperty
    public int pageSize;

    @JsonProperty
    public int resultCount;

    @JsonProperty
    public int totalCount;
}

File: src/main/kotlin/com/undsf/mcmodmgr/curseforge/responses/Pagination.kt
package com.undsf.mcmodmgr.curseforge.responses

class Pagination(
        var index: Int,
        var pageSize: Int,
        var resultCount: Int,
        var totalCount: Int) {
    fun endIndex() : Int {
        return index + resultCount
    }

    fun hasNextPage() : Boolean {
        return endIndex() < totalCount
    }
}